### DISCUSSION TITLE
Accessing inner ContentTabs object from TabbedContent

### DISCUSSION THREAD

**Discussion by scmanjarrez** [Textualize/textual#4808] [Q&A] [answered]
Hi, I'm trying to create a TUI with a TabbedContent and some Input widgets, when the Input is focused I can't switch between tabs anymore with arrows, so my first idea was to change the focus back to the container of my tabs, however, focusing TabbedContent doesn't work. After some debugging I've noticed that the previously focused widget was ContentTabs, but I don't see a function nor attribute to access it. In the source code I see calls `self.get_child_by_type(ContentTabs)`, but that doesn't work here because ContentTabs isn't a public object

Edit: Nevermind, I can access it by importing the private module `from textual.widgets._tabbed_content import ContentTabs`

![Peek 2024-07-26 15-38](https://github.com/user-attachments/assets/6a513009-02bb-4d96-9fe8-49fd095c20fc)

```python
from textual.app import App, ComposeResult
from textual.widgets import Footer, TabbedContent, Markdown, TabPane, Input


class TestApp(App):
    BINDINGS = [
        ("q", "quit", "Quit de app"),
        ("u", "first", "First field"),
        ("r", "second", "Second field"),
    ]

    def compose(self) -> ComposeResult:
        with TabbedContent(classes="box", id="tabcont"):
            with TabPane("Tab 1"):
                yield Markdown("# Pane 1")
                yield Input(placeholder="field 1")
                yield Input(placeholder="field 2")
            with TabPane("Tab 2"):
                yield Markdown("# Pane 2")
        yield Footer()

    def on_key(self, event):
        if event.key == "escape":
            prev = self.app.screen.focus_previous()
            print(prev)
            print(type(prev))
            tc = self.query_one(TabbedContent)
            print(tc)
            print(type(tc))


if __name__ == '__main__':
    app = TestApp()
    app.run()
```

<details>
  <summary>P.S.: Is it possible to prevent clicks on the footer generate key events? Edit: This can be prevented by using [check_action](https://textual.textualize.io/guide/actions/#dynamic-actions)</summary>
  
![Peek 2024-07-26 15-38-2](https://github.com/user-attachments/assets/68bf44af-fb07-4639-9aaf-0d52df27c37e)
  
</details>

**Comment by TomJGooding** [comment, marked as answer]
`ContentTabs`  is now publicly exported after Textual v0.73.0 - see #4765

**Reply by scmanjarrez** [reply]
Oh, thank you! I was importing it directly from widgets, but it seems that I need to import it from tabbed_content `from textual.widgets.tabbed_content import ContentTabs`
